src/core/resources/ens/ensRegistration.ts

```diff
@@ -0,0 +1,111 @@
+import { keccak256 } from '@ethersproject/keccak256';
+import { toUtf8Bytes } from '@ethersproject/strings';
+import { useQuery } from '@tanstack/react-query';
+import uts46 from 'idna-uts46-hx';
+
+import { ensClient } from '~/core/graphql';
+import {
+  QueryConfig,
+  QueryFunctionArgs,
+  QueryFunctionResult,
+  createQueryKey,
+} from '~/core/react-query';
+
+const ENS_DOMAIN = '.eth';
+
+function normalizeENS(name: string) {
+  try {
+    return uts46.toUnicode(name, { useStd3ASCII: true });
+  } catch (err) {
+    return name;
+  }
+}
+
+function decodeLabelhash(hash: string) {
+  if (!(hash.startsWith('[') && hash.endsWith(']'))) {
+    throw Error(
+      'Expected encoded labelhash to start and end with square brackets',
+    );
+  }
+
+  if (hash.length !== 66) {
+    throw Error('Expected encoded labelhash to have a length of 66');
+  }
+
+  return `${hash.slice(1, -1)}`;
+}
+
+function isEncodedLabelhash(hash: string) {
+  return hash.startsWith('[') && hash.endsWith(']') && hash.length === 66;
+}
+
+function labelhash(unnormalisedLabelOrLabelhash: string) {
+  if (unnormalisedLabelOrLabelhash === '[root]') {
+    return '';
+  }
+  return isEncodedLabelhash(unnormalisedLabelOrLabelhash)
+    ? '0x' + decodeLabelhash(unnormalisedLabelOrLabelhash)
+    : keccak256(toUtf8Bytes(normalizeENS(unnormalisedLabelOrLabelhash)));
+}
+
+const fetchRegistration = async (ensName: string) => {
+  const response = await ensClient.getRegistration({
+    id: labelhash(ensName.replace(ENS_DOMAIN, '')),
+  });
+  const data = response.registration;
+
+  return {
+    registration: {
+      expiryDate: data?.expiryDate as string,
+      registrationDate: data?.registrationDate as string,
```

Comment:
I think this type technically is `string | undefined`, is the `as string` needed? afraid someone assumes it always has a value in the future

Reply:
i agree with this, we should have a fallback or nothing at all (?) if this fails

Reply:
Yep, this makes sense. I initially did this because these gql values are assigned as `any`. I have now changed this type to `string | undefined`.